controllers/gce/loadbalancers/loadbalancers.go

```diff
@@ -872,7 +867,7 @@ func (l *L7) Cleanup() error {
 		l.tps = nil
 	}
 	// Delete the SSL cert if it is not a pre-created GCE cert.
-	if l.sslCert != nil && l.sslCert.Name != l.runtimeInfo.TLSName {
+	if l.sslCert != nil && l.runtimeInfo.TLSName != "" {
```

Comment:
Wouldn't this do the opposite? TLSName is non-empty when a pre-shared cert is specified. We want this code block to run when a ssl-cert was created by a secret.